backend/compact-connect/lambdas/python/common/cc_common/data_model/data_client.py

```diff
@@ -2733,6 +2754,27 @@ def encumber_home_jurisdiction_license_privileges(
             )
 
         for encumbered_privilege in previously_encumbered_privileges_associated_with_license:
+            # Check if an update record already exists for this adverse action
+            # to avoid creating duplicate update records if the event flow is re-run
+            existing_updates = provider_user_records.get_update_records_for_privilege(
+                jurisdiction=encumbered_privilege.jurisdiction,
+                license_type=encumbered_privilege.licenseType,
+                filter_condition=lambda update: (
+                    update.updateType == UpdateCategory.ENCUMBRANCE
+                    and update.encumbranceDetails is not None
+                    and str(update.encumbranceDetails.get('adverseActionId')) == adverse_action_id
```

Comment:
We've been working on tidying up our object type handling throughout the codebase. When we have a UUID, we should deserialize it early and handle it as a `UUID`, rather than a `str`, only serializing again to a `str` as late as possible before it leaves the system. Similarly, our date/datetime objects should be processed as the more intelligent types, not strings. To that end, rather than cast `adverseActionId` here to a `str,` can you instead chase this `adverse_action_id` up the call chain and convert it to a UUID where it comes in?